helix-term/src/ui/editor.rs

```diff
@@ -171,8 +187,19 @@ impl EditorView {
             } else {
                 Box::new(syntax::merge(highlights, focused_view_elements))
             }
+        };
+
+        let highlights = if view.in_visual_jump_mode {
```

Comment:
We already have a concept of "jump", which is different from the jump mode that this PR is implementing. So to avoid confusion, maybe it's better to rename it to something like hop mode? (this PR is heavily influenced by hop.nvim anyways..)

Reply:
perhaps `hone` or `skip`? a different name would certainly make searching for the feature/related-code easier